src/renderer/pages/Basket/ui/SignOperations.tsx

```diff
@@ -0,0 +1,173 @@
+  UnstakeConfirmation,
+  WithdrawConfirmation,
+} from '@features/operations/OperationsConfirm';
+import { TransferTypes, XcmTypes } from '@entities/transaction';
+import { networkModel } from '@/src/renderer/entities/network';
+import { OperationTitle } from '@/src/renderer/entities/chain';
+import { getOperationTitle } from '../lib/operation-title';
+import { cnTw } from '@/src/renderer/shared/lib/utils';
+
+export const SignOperations = () => {
+  const { t } = useI18n();
+
+  const step = useUnit(signOperationsModel.$step);
+  const transactions = useUnit(signOperationsModel.$transactions);
+  const chains = useUnit(networkModel.$chains);
+
+  const [currentTx, setCurrentTx] = useState(0);
+
+  const [isModalOpen, closeModal] = useModalClose(
+    !signOperationsUtils.isNoneStep(step),
+    signOperationsModel.output.flowFinished,
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const scroll = (value: number) => {
+    setTimeout(() =>
+      // TS doesn't recognize offsetLeft
+      // @ts-ignore
+      ref.current?.scrollTo({ left: ref.current?.childNodes[0].childNodes[value].offsetLeft - 16, behavior: 'smooth' }),
+    );
+  };
+
+  if (signOperationsUtils.isSubmitStep(step)) return <OperationSubmit isOpen={isModalOpen} onClose={closeModal} />;
+
+  const getModalTitle = (basketTransaction: BasketTransaction): String | ReactNode => {
+    const chain = chains[basketTransaction.coreTx.chainId];
+
+    const { title, params } = getOperationTitle(basketTransaction, chain);
+
+    return (
+      <OperationTitle
+        className="justify-center my-3"
+        title={`${t(title, { ...params })}`}
+        chainId={basketTransaction.coreTx.chainId}
+      />
+    );
+  };
+
+  const getConfirmScreen = (transaction: BasketTransaction) => {
+    const type = transaction.coreTx.type;
+
+    if (TransferTypes.includes(type) || XcmTypes.includes(type)) {
+      return () => <TransferConfirm hideSignButton />;
+    }
+
+    const Components = {
+      // Proxy
+      [TransactionType.ADD_PROXY]: () => <AddProxyConfirm hideSignButton />,
+      [TransactionType.REMOVE_PROXY]: () => <RemoveProxyConfirm hideSignButton />,
+      [TransactionType.CREATE_PURE_PROXY]: () => <AddPureProxiedConfirm hideSignButton />,
+      [TransactionType.REMOVE_PURE_PROXY]: () => <RemovePureProxiedConfirm hideSignButton />,
+      // Staking
+      [TransactionType.BOND]: () => <BondNominateConfirmation hideSignButton />,
+      [TransactionType.NOMINATE]: () => <NominateConfirmation hideSignButton />,
+      [TransactionType.STAKE_MORE]: () => <BondExtraConfirmation hideSignButton />,
+      [TransactionType.REDEEM]: () => <WithdrawConfirmation hideSignButton />,
+      [TransactionType.RESTAKE]: () => <RestakeConfirmation hideSignButton />,
+      [TransactionType.DESTINATION]: () => <PayeeConfirmation hideSignButton />,
+      [TransactionType.UNSTAKE]: () => <UnstakeConfirmation hideSignButton />,
+    };
+
+    // @ts-ignore
+    return Components[type];
```

Comment:
why do we need `ts-ignore` here? We don't use all of the `TransactionType` here or is it something else?

Reply:
Right, we don't use multisig, proxy and technical txs like chill or batch